src/main/java/com/shaft/gui/element/internal/ElementActionsHelper.java

```diff
@@ -819,6 +827,115 @@ public static String typeWrapper(ElementInformation elementInformation, String t
             return adjustedTargetText;
         }
     }
+    public static String readElementText(ElementInformation elementInformation) {
+
+          //  var elementInformation = ElementInformation.fromList(ElementActionsHelper.identifyUniqueElementIgnoringVisibility(DriverFactoryHelper.getDriver(), elementLocator));
+            String elementText;
+            try {
+                elementText = (elementInformation.getFirstElement()).getText();
+            } catch (WebDriverException webDriverException) {
+                elementText = ElementInformation.fromList(ElementActionsHelper.performActionAgainstUniqueElementIgnoringVisibility(DriverFactoryHelper.getDriver(), elementInformation.getLocator(), ElementAction.GET_TEXT)).getActionResult();
+            }
+            if ((elementText == null || elementText.isBlank()) && !DriverFactoryHelper.isMobileNativeExecution()) {
+                try {
+                    elementText = (elementInformation.getFirstElement()).getAttribute(ElementActionsHelper.TextDetectionStrategy.CONTENT.getValue());
+                } catch (WebDriverException webDriverException) {
+                    elementText = ElementInformation.fromList(ElementActionsHelper.performActionAgainstUniqueElementIgnoringVisibility(DriverFactoryHelper.getDriver(), elementInformation.getLocator(), ElementAction.GET_CONTENT)).getActionResult();
+                }
+            }
+            if ((elementText == null || elementText.isBlank()) && !DriverFactoryHelper.isMobileNativeExecution()) {
+                try {
+                    elementText = (elementInformation.getFirstElement()).getAttribute(ElementActionsHelper.TextDetectionStrategy.VALUE.getValue());
+                } catch (WebDriverException webDriverException) {
+                    elementText = ElementInformation.fromList(ElementActionsHelper.performActionAgainstUniqueElementIgnoringVisibility(DriverFactoryHelper.getDriver(), elementInformation.getLocator(), ElementAction.GET_VALUE)).getActionResult();
+                }
+            }
+            if (elementText == null) {
+                elementText = "";
+            }
+            return elementText;
+    }
+    private static void newClearBeforeTyping(By elementLocator){
+        var elementInformation = ElementInformation.fromList(ElementActionsHelper.performActionAgainstUniqueElement(DriverFactoryHelper.getDriver(), elementLocator));
+        var attemptClearBeforeTyping = SHAFT.Properties.flags.attemptClearBeforeTyping();
+        var attemptClearBeforeTypingUsingBackspace = SHAFT.Properties.flags.attemptClearBeforeTypingUsingBackspace();
+        if (attemptClearBeforeTyping && attemptClearBeforeTypingUsingBackspace) {
+            clearBeforeTypingUsingBackSpace(elementInformation);
+        }
+        else if (attemptClearBeforeTyping) {
+            clearBeforeTypingUsingNativeClear(elementInformation);
+        }
+        else if(attemptClearBeforeTypingUsingBackspace){
```

Comment:
I think this part is redundant.
My suggestion this so it wont go to the condition if the `attemptClearBeforeTyping ` property is disabled from the beginning:

```
if (attemptClearBeforeTyping) {
        if (attemptClearBeforeTypingUsingBackspace) {
             clearBeforeTypingUsingBackSpace(elementInformation);
        }
        else {
              clearBeforeTypingUsingNativeClear(elementInformation);
        }
}
```

Reply:
> I think this part is redundant. My suggestion this so it wont go to the condition if the `attemptClearBeforeTyping ` property is disabled from the beginning:
> 
> ```
> if (attemptClearBeforeTyping) {
>         if (attemptClearBeforeTypingUsingBackspace) {
>              clearBeforeTypingUsingBackSpace(elementInformation);
>         }
>         else {
>               clearBeforeTypingUsingNativeClear(elementInformation);
>         }
> }
> ```

@MahmoudElSharkawy if the attemptToclearBeforeTyping is false and using back space is true then no clearing will happen , i think we should seprate flags from each other , what do u think

Reply:
done